webapp/src/ts/services/xml-forms.service.ts

```diff
@@ -162,6 +162,9 @@ export class XmlFormsService {
       if (!validSoFar) {
         return false;
       }
+      if (form.context.permission) {
+        return this.authService.has(form.context.permission);
```

Comment:
Returning here means that if we have a permission, we never check the expression?